frontend/src/contexts/HostContext/HostContext.spec.js

```diff
@@ -141,136 +141,342 @@ describe('Context', () => {
       expect(screen.getByTestId('lobby-id')).toHaveTextContent('foo');
     });
 
-    it('catches player-connected events and dispatches its respective action', () => {
-      const TestComponent = () => {
-        const { state } = useContext(HostContext);
+    describe('when the host receives a player-connected event', () => {
+      it('adds the player to player staging if their name is not already in staging', () => {
+        const TestComponent = () => {
+          const { state } = useContext(HostContext);
+
+          return (
+            <>
+              <div data-testid="game-state">{state.gameState}</div>
+              <div data-testid="lobby-id">{state.lobbyID}</div>
+
+              <div>
+                {state.newPlayerStaging.map(() => (
+                  <span data-testid="player-staging" />
+                ))}
+              </div>
+            </>
+          );
+        };
+
+        const { eventHandlers } = setupEmitterMocks();
+
+        render(
+          <HostProvider>
+            <TestComponent />
+          </HostProvider>,
+        );
 
-        return (
-          <>
-            <div data-testid="game-state">{state.gameState}</div>
-            <div data-testid="lobby-id">{state.lobbyID}</div>
+        expect(screen.queryAllByTestId('player-staging').length).toBe(0);
 
-            <div>
-              {state.newPlayerStaging.map(() => (
-                <span data-testid="player-staging" />
-              ))}
-            </div>
-          </>
+        act(() => {
+          eventHandlers.message({
+            event: 'player-connected',
+            payload: { playerId: 'TEST', playerName: 'TESTER' },
+          });
+        });
+
+        expect(screen.queryAllByTestId('player-staging').length).toBe(1);
+      });
+
```

Comment:
Note to make reviewing easier: I didn't change anything here other than the test name for matching my update and moving it into another describe block to group all of these (now many) `player-connected` related tests together.